Cesium3DTilesSelection/src/WebMapServiceRasterOverlay.cpp

```diff
@@ -0,0 +1,410 @@
+
+    const CesiumGeospatial::GlobeRectangle tileRectangle =
+        CesiumGeospatial::unprojectRectangleSimple(this->getProjection(), options.rectangle);
+
+    const std::string urlTemplate =
+        "{baseUrl}?request=GetMap&TRANSPARENT=TRUE&version={version}&service=wms&"
+        "format=image.png&styles="
+        "&width={width}&height={height}&bbox={minx},{miny},{maxx},{maxy}"
+        "&layers={layername}&crs=EPSG:4326";
+
+    const auto radiansToDegrees = [](double rad) { 
+        return std::to_string(CesiumUtility::Math::radiansToDegrees(rad));
+    };
+
+    const std::map<std::string, std::string> urlTemplateMap = { 
+      {"baseUrl", this->_url},
+      {"version", this->_version},
+      {"maxx", radiansToDegrees(tileRectangle.getNorth())},
+      {"maxy", radiansToDegrees(tileRectangle.getEast())},
+      {"minx", radiansToDegrees(tileRectangle.getSouth())},
+      {"miny", radiansToDegrees(tileRectangle.getWest())},
+      {"layername", this->_layername},
+      {"width", std::to_string(this->getWidth())},
+      {"height", std::to_string(this->getHeight())}
+    };
+
+    std::string url = CesiumUtility::Uri::substituteTemplateParameters(
+        urlTemplate,
+        [
+          map=&urlTemplateMap
+        ](const std::string& placeholder) {
+          return map->count(placeholder) == 0 ? "{" + placeholder + "}"
+                                              : map->at(placeholder);
+        }); 
+
+    //SPDLOG_LOGGER_INFO(getLogger(), url);
+
+    return this->loadTileImageFromUrl(url, this->_headers, std::move(options));
+  }
+
+private:
+  std::string _url;
+  std::vector<IAssetAccessor::THeader> _headers;
+  std::string _version;
+  std::string _layername;
+  std::string _fileExtension;
+};
+
+WebMapServiceRasterOverlay::WebMapServiceRasterOverlay(
+    const std::string& name,
+    const std::string& url,
+    const std::vector<IAssetAccessor::THeader>& headers,
+    const WebMapServiceRasterOverlayOptions& wmsOptions,
+    const RasterOverlayOptions& overlayOptions)
+    : RasterOverlay(name, overlayOptions),
+      _baseUrl(url),
+      _headers(headers),
+      _options(wmsOptions) {} 
+
+WebMapServiceRasterOverlay::~WebMapServiceRasterOverlay() {}
+
+static bool validateCapabilities(tinyxml2::XMLElement* pRoot, const WebMapServiceRasterOverlayOptions& options, std::string& error) {
+  tinyxml2::XMLElement* pService = pRoot->FirstChildElement("Service");
+  if (!pService) {
+    error = "Web map service XML document does not have a Service "
+            "element. ";
+    return false;
+  }
+
+  tinyxml2::XMLElement* pServiceName =
+      pService->FirstChildElement("Name");
+  if (!pServiceName) {
+    error = "Invalid web map service XML document (Service > Name is "
+            "missing) ";
+    return false;
+  }
+
+  const std::string serviceNameText = pServiceName->GetText();
+  if (serviceNameText != "WMS") {
```

Comment:
There is no requirement that the service be named `WMS`. Servers can return whatever they like here.